### PR TITLE
#2 explicit dependecies in esn account

### PR DESCRIPTION
**Pull request by asmaNode**

### CODE REVIEW

File: src/linagora.esn.controlcenter/app/app.module.js [hunk 1]
```diff
@@ -12,7 +25,6 @@
     'esn.user-configuration',
     'esn.business-hours',
     'esn.configuration',
-    'esn.user-configuration',
     'esn.i18n',
     'esn.home-page'
```
Comment (MichaelBailly) [inline src/linagora.esn.controlcenter/app/app.module.js:29]: you need to add the dependency to the application menu (`esn.application-menu`)

File: src/linagora.esn.controlcenter/app/general/controlcenter-general.controller.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const _ = require('lodash');
```
Comment (MichaelBailly) [inline src/linagora.esn.controlcenter/app/general/controlcenter-general.controller.js:1]: You just added the _ provided by webpack. You should remove the _ provided through AngularJS injection (
```
function controlcenterGeneralController(
    $q,
    $window,
    esnUserConfigurationService,
    asyncAction,
    homePageService,
    _, // <- REMOVE HERE
    ESN_ROUTER_DEFAULT_HOME_PAGE,
    CONTROLCENTER_GENERAL_CONFIGS
  ) {
```

File: src/linagora.esn.profile/app/app.module.js [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+require('esn-frontend-common-libs/src/frontend/js/modules/session.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/profile.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/notification.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/timeline/timeline.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/previous-page.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/i18n/i18n.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/async-action.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/availability/availability.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/module-registry/module-registry.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/user-configuration/user-configuration.module.js');
```
Comment (MichaelBailly) [inline src/linagora.esn.profile/app/app.module.js:16]: You need to add the associated AngularJS dependency for user-configuration.module.js

File: src/linagora.esn.profile/app/edit/profile-edit.controller.js [hunk 4]
```diff
@@ -1,3 +1,6 @@
+require('../common/profile-api.service.js')
+const _ = require('lodash');
```
Comment (MichaelBailly) [inline src/linagora.esn.profile/app/edit/profile-edit.controller.js:2]: remove the Angular injected `_`

File: src/linagora.esn.controlcenter/app/app.module.js (hunk 1 above)
Comment (asmaNode) [reply src/linagora.esn.controlcenter/app/app.module.js]: fixed!

File: src/linagora.esn.controlcenter/app/app.module.js [hunk 5]
```diff
@@ -1,3 +1,16 @@
...
+require('esn-frontend-common-libs/src/frontend/js/modules/header/header.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/user/user.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/sidebar.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/module-registry/module-registry.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/user-configuration/user-configuration.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/business-hours/business-hours.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/config/config.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/i18n/i18n.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/home-page/home-page.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/application-menu.js');
```
Comment (tuanlc) [inline src/linagora.esn.controlcenter/app/app.module.js:12]: So, I see you collect all the requirements in the app.module.js file.

Comment (asmaNode) [reply src/linagora.esn.controlcenter/app/app.module.js:12]: YES, based on the comment of Christophe I must  put all requirement in root module

File: src/linagora.esn.controlcenter/app/app.module.js [hunk 6]
```diff
@@ -1,3 +1,16 @@
...
+
```
Comment (chamerling) [inline src/linagora.esn.controlcenter/app/app.module.js:13]: So, you must `require` all the local files from this `linagora.esn.account` module

File: src/linagora.esn.profile/app/app.module.js [hunk 7]
```diff
@@ -1,3 +1,20 @@
...
+
```
Comment (chamerling) [inline src/linagora.esn.profile/app/app.module.js:17]: Same here, `require` all the local JS files

File: src/index.js [hunk 8]
```diff
@@ -1,51 +1,27 @@
...
+require('esn-frontend-common-libs/src/frontend/js/modules/core.js');
+require('esn-frontend-common-libs/src/frontend/components/dynamic-directive/src/dynamic-directive.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/http.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/session.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/profile.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/notification.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/timeline/timeline.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/previous-page.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/async-action.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/availability/availability.module.js');
```
Comment (MichaelBailly) [inline src/index.js:24]: I do not agree with this. The "esn-frontend-common-libs" dependencies should be required in the modules definition files, which are `./linagora.esn.controlcenter/app/app.module.js` and `./linagora.esn.profile/app/app.module.js`.

The index.js file can then be:

```javascript
require('esn-frontend-common-libs/src/index.js');
require('./require-angular-injections.js');

require('./linagora.esn.controlcenter/app/app.module.js');
require('./linagora.esn.profile/app/app.module.js');
```
The issue you had in your chat comment "when  removing local dependencies from index js and put it in app.module.js  , the application crash" was most likely due to you requiring a module directive|service|controller before the module definition, like for example:

```javascript
// DON'T DO THAT
require ('./linagora.esn.controlcenter/app/app.route.js'); // module doesn't exist yet and we try to add something inside
require('./linagora.esn.controlcenter/app/app.module.js'); // module is created here
```
Order matters, and webpack cannot order it automatically.

**Pull request merged**